src/uwtools/tests/test_config.py

```diff
@@ -0,0 +1,861 @@
+    cfgobj.dump_file(reference)
+    assert compare_files(reference, outfile)
+
+
+# Tests
+
+
+def test_bad_conversion_cfg_to_pdf(capsys):
+    with raises(SystemExit):
+        config.create_config_obj(
+            parse_config_args(["-i", fixture_path("simple2_nml.cfg"), "--input-file-type", ".pdf"])
+        )
+    assert "invalid choice: '.pdf'" in capsys.readouterr().err
+
+
+# def test_bad_conversion_depth_mismatch(tmp_path):
+#     cfgfile = str(tmp_path / "cfg.yaml")
+#     outfile = str(tmp_path / "out.nml")
+#     with open(cfgfile, "w", encoding="utf-8") as f:
+#         print("{a: {b: {c: 88}}}", file=f)
+#     with raises(ValueError):  # as e:
+#         config.create_config_obj(parse_config_args(["-i", cfgfile, "-o", outfile]))
+#     # assert something
+
+
+def test_bad_conversion_nml_to_yaml():
+    with raises(ValueError):
+        config.create_config_obj(
+            parse_config_args(
+                [
+                    "-i",
+                    fixture_path("simple2.nml"),
+                    "-c",
+                    fixture_path("srw_example.yaml"),
+                    "--config-file-type",
+                    "YAML",
+                ]
+            )
+        )
+
+
+def test_bad_conversion_yaml_to_nml(tmp_path):
+    with raises(ValueError):
+        config.create_config_obj(
+            parse_config_args(
+                [
+                    "-i",
+                    fixture_path("srw_example.yaml"),
+                    "-o",
+                    str(tmp_path / "test_outfile_conversion.yaml"),
+                    "--output-file-type",
+                    "F90",
+                ]
+            )
+        )
+
+
+def test_cfg_to_yaml_conversion(tmp_path):
+    """
+    Test that a .cfg file can be used to create a YAML object.
+    """
+    infile = fixture_path("srw_example_yaml.cfg")
+    outfile = str(tmp_path / "test_ouput.yaml")
+    config.create_config_obj(
+        parse_config_args(["-i", infile, "-o", outfile, "--input-file-type", "YAML"])
+    )
+    expected = config.YAMLConfig(infile)
+    expected.dereference_all()
+    expected_file = tmp_path / "test.yaml"
+    expected.dump_file(expected_file)
+    assert compare_files(expected_file, outfile)
+    with open(outfile, "r", encoding="utf-8") as f:
+        assert f.read()[-1] == "\n"
+
+
+@pytest.mark.parametrize("fmt", ["F90", "INI", "YAML"])
+def test_compare_config(fmt, salad_base, caplog):
+    """Compare two config objects."""
+    ext = ".%s" % ("nml" if fmt == "F90" else fmt).lower()
+    cfgobj = help_cfgclass(ext)(fixture_path(f"simple{ext}"), log_name="test_compare_config")
+    cfgobj.log.addHandler(logging.StreamHandler(sys.stdout))
+    cfgobj.log.setLevel(logging.INFO)
+    cfgobj.compare_config(cfgobj, salad_base)
+    if fmt in ["F90", "YAML"]:
+        assert msg_in_caplog("salad:        how_many:  - 12 + 12", caplog.records)
+    else:
+        assert not caplog.records  # #PM# BUT WHY NOT IN INI?
```

Comment:
I have none. It's not ringing any bells.

Reply:
@christinaholtNOAA I found it: For F90 and YAML, the `"how_many": "12"` `str` value in the `assets` test (inline `pytest`) fixture is a mismatch with the `12` `int` value in the (external file) fixture, but for INI -- which I assume parses everything a `str` objects? -- it's a match. So I'm updating the tests that use this fixture so that we don't have this weird edge case.